core/includes/date.inc

```diff
@@ -450,6 +450,44 @@ function date_format_patterns($strict = FALSE) {
   );
 }
 
+/**
+ * Creates an array of date format types for use as an options list.
+ *
+ * @param bool $html
+ *   Whether to include HTML date options or not.
+ *
+ * @return array
+ *   All date format options, keyed by machine name.
+ */
+function date_format_type_options($html = TRUE) {
```

Comment:
Hm. Is that API change really necessary? Function date_format_type_options() currently has no params.